src/nvim/api/options.c

```diff
@@ -114,6 +114,9 @@ static buf_T *do_ft_buf(char *filetype, aco_save_T *aco, Error *err)
     return NULL;
   }
 
+  // Setting buffer filetype in other situations loads the buffer
+  buf_ensure_loaded(ftbuf);
```

Comment:
I'm still not convinced why this one is needed. It causes extra disk IO and getting filetype options is already performance sensitive.